index.js

```diff
@@ -40,8 +40,11 @@ class ExecutorRouter extends Executor {
             const options = Object.assign({ ecosystem }, plugin.options); // Add ecosystem to executor options
             const executorPlugin = new ExecutorPlugin(options);
 
-            // Set the default executor
-            if (!this.defaultExecutor && !config.defaultPlugin) {
+            if (defaultPlugin && defaultPlugin === plugin.name) {
```

Comment:
If you specified a `defaultPlugin`, it shouldn't matter what order in the array that resides in.

If you didn't specify a `defaultPlugin`, the `this.defaultExecutor` will be the first plugin we are able to load.